hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -674,6 +698,17 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			log.Error(fmt.Errorf("release image is invalid"), "reconciliation is blocked", "message", validReleaseImage.Message)
 			return ctrl.Result{}, nil
 		}
+		if hcluster.Status.Version != nil && hcluster.Status.Version.Desired.Image != hcluster.Spec.Release.Image {
```

Comment:
note aside: we should have similar mechanism to prevent downgrades.

Reply:
Outside this PR but yes.  We have protection against y-stream downgrade in `isValidReleaseVersion` but not z-stream.  Not sure if z-stream downgrade requires a force or not.  Need to look into it.